### ISSUE TITLE
Extension not working - Login Screen does not open

### ISSUE THREAD

**Issue by bunducafe** [marcelklehr/floccus#61]
Well, the title says it all.

After installing the extension for the latest firefox version, a click on the icon does literally nothing. It opens a small white space where I cannot click or write anything into it.

I am on the latest macOS with the latest version of Firefox and nextcloud 13.

Any hints?

**Comment by marcelklehr** [comment]
Hey @bunducafe 
I'm sorry you're having trouble. You could try gathering debug logs as follows:

#### Firefox
* Go to `about:debugging`
* enable debuggingng
* click on `debug` button next to floccus entry and confirm the dialog
* open floccus options
* go back to the debugger window and copy the logs

**Comment by bunducafe** [comment]
Hey @marcelklehr 

I tried to uninstall and reinstall the Extension and at least I could then enter my server details. Unfortunately with the second click on the extension I am getting the same error that I cannot see anything, so the popup does not show up.

That's what the console gathers after clicking on the icon in order to open the options:


          Die Zeichenkodierung des HTML-Dokuments wurde nicht deklariert. Das Dokument wird in manchen Browser-Konfigurationen mit verstümmeltem Text dargestellt, wenn das Dokument Zeichen außerhalb des US-ASCII-Bereichs enthält. Die Zeichenkodierung der Seite muss im Dokument oder Transferprotokoll deklariert werden.  background.html
Error: Permission denied to access property "a"[Weitere Informationen]  common.js:1:1
error occurred while processing 'sources: TypeError: window is null
Stack: WebExtensionChildActor.prototype.isExtensionWindow@resource://devtools/shared/base-loader.js -> resource://devtools/server/actors/webextension.js:272:3
WebExtensionChildActor.prototype._allowSource@resource://devtools/shared/base-loader.js -> resource://devtools/server/actors/webextension.js:292:13
TabSources/this.allowSource@resource://devtools/shared/base-loader.js -> resource://devtools/server/actors/utils/TabSources.js:32:39
source@resource://devtools/shared/base-loader.js -> resource://devtools/server/actors/utils/TabSources.js:120:12
createNonSourceMappedActor@resource://devtools/shared/base-loader.js -> resource://devtools/server/actors/utils/TabSources.js:359:12
createSourceActors/<@resource://devtools/shared/base-loader.js -> resource://devtools/server/actors/utils/TabSources.js:401:19
process@resource://gre/modules/Promise.jsm -> resource://gre/modules/Promise-backend.js:922:23
walkerLoop@resource://gre/modules/Promise.jsm -> resource://gre/modules/Promise-backend.js:806:7
scheduleWalkerLoop/<@resource://gre/modules/Promise.jsm -> resource://gre/modules/Promise-backend.js:742:11
enter@resource://devtools/shared/base-loader.js -> resource://devtools/server/actors/script.js:356:5
_pushThreadPause@resource://devtools/shared/base-loader.js -> resource://devtools/server/actors/script.js:539:5
onExceptionUnwind@resource://devtools/shared/base-loader.js -> resource://devtools/server/actors/script.js:1898:7
[188]</<@moz-extension://f3feffe7-1653-cd48-86d9-690b0072a16b/dist/js/polyfill.js:3993:3
[35]</module.exports@moz-extension://f3feffe7-1653-cd48-86d9-690b0072a16b/dist/js/polyfill.js:825:14
[188]<@moz-extension://f3feffe7-1653-cd48-86d9-690b0072a16b/dist/js/polyfill.js:3991:7
s@moz-extension://f3feffe7-1653-cd48-86d9-690b0072a16b/dist/js/polyfill.js:1:246
s/<@moz-extension://f3feffe7-1653-cd48-86d9-690b0072a16b/dist/js/polyfill.js:1:305
[327]<@moz-extension://f3feffe7-1653-cd48-86d9-690b0072a16b/dist/js/polyfill.js:6660:1
s@moz-extension://f3feffe7-1653-cd48-86d9-690b0072a16b/dist/js/polyfill.js:1:246
s/<@moz-extension://f3feffe7-1653-cd48-86d9-690b0072a16b/dist/js/polyfill.js:1:305
[1]</<@moz-extension://f3feffe7-1653-cd48-86d9-690b0072a16b/dist/js/polyfill.js:5:1
[1]<@moz-extension://f3feffe7-1653-cd48-86d9-690b0072a16b/dist/js/polyfill.js:2:2
s@moz-extension://f3feffe7-1653-cd48-86d9-690b0072a16b/dist/js/polyfill.js:1:246
e@moz-extension://f3feffe7-1653-cd48-86d9-690b0072a16b/dist/js/polyfill.js:1:425
@moz-extension://f3feffe7-1653-cd48-86d9-690b0072a16b/dist/js/polyfill.js:1:11
Line: 272, column: 3  main.js:1651
Content Security Policy: 'x-frame-options' wird wegen 'frame-ancestors'-Direktive ignoriert.  (unbekannt)
Content Security Policy: Die Direktive 'child-src' sollte nicht mehr verwendet werden. Bitte verwenden Sie stattdessen die Direktive 'worker-src' zum Kontrollieren von Workern bzw. die Direktive 'frame-src' zum Kontrollieren von Frames.  (unbekannt)
Content Security Policy: Die Einstellungen der Seite haben das Laden einer Ressource auf self blockiert ("script-src"). Source: (function(a,w,v,I){var c={safeWindow:{},....  61:1
Content Security Policy: Die Einstellungen der Seite haben das Laden einer Ressource auf self blockiert ("script-src"). Source: (() => {
        "use strict";
        c.... 61:1
Content Security Policy: Die Einstellungen der Seite haben das Laden einer Ressource auf self blockiert ("script-src"). Source: (() => {

        "use strict";



     ....  61:1
Content Security Policy: Die Einstellungen der Seite haben das Laden einer Ressource auf self blockiert ("script-src"). Source: (() => {

        "use strict";

       .... 61:1
error occurred while processing 'sources: TypeError: window is null
Stack: WebExtensionChildActor.prototype.isExtensionWindow@resource://devtools/shared/base-loader.js -> resource://devtools/server/actors/webextension.js:272:3
WebExtensionChildActor.prototype._allowSource@resource://devtools/shared/base-loader.js -> resource://devtools/server/actors/webextension.js:292:13
TabSources/this.allowSource@resource://devtools/shared/base-loader.js -> resource://devtools/server/actors/utils/TabSources.js:32:39
source@resource://devtools/shared/base-loader.js -> resource://devtools/server/actors/utils/TabSources.js:120:12
createNonSourceMappedActor@resource://devtools/shared/base-loader.js -> resource://devtools/server/actors/utils/TabSources.js:359:12
createSourceActors/<@resource://devtools/shared/base-loader.js -> resource://devtools/server/actors/utils/TabSources.js:401:19
process@resource://gre/modules/Promise.jsm -> resource://gre/modules/Promise-backend.js:922:23
walkerLoop@resource://gre/modules/Promise.jsm -> resource://gre/modules/Promise-backend.js:806:7
scheduleWalkerLoop/<@resource://gre/modules/Promise.jsm -> resource://gre/modules/Promise-backend.js:742:11
enter@resource://devtools/shared/base-loader.js -> resource://devtools/server/actors/script.js:356:5
_pushThreadPause@resource://devtools/shared/base-loader.js -> resource://devtools/server/actors/script.js:539:5
onExceptionUnwind@resource://devtools/shared/base-loader.js -> resource://devtools/server/actors/script.js:1898:7
[188]</<@moz-extension://f3feffe7-1653-cd48-86d9-690b0072a16b/dist/js/polyfill.js:3993:3
[35]</module.exports@moz-extension://f3feffe7-1653-cd48-86d9-690b0072a16b/dist/js/polyfill.js:825:14
[188]<@moz-extension://f3feffe7-1653-cd48-86d9-690b0072a16b/dist/js/polyfill.js:3991:7
s@moz-extension://f3feffe7-1653-cd48-86d9-690b0072a16b/dist/js/polyfill.js:1:246
s/<@moz-extension://f3feffe7-1653-cd48-86d9-690b0072a16b/dist/js/polyfill.js:1:305
[327]<@moz-extension://f3feffe7-1653-cd48-86d9-690b0072a16b/dist/js/polyfill.js:6660:1
s@moz-extension://f3feffe7-1653-cd48-86d9-690b0072a16b/dist/js/polyfill.js:1:246
s/<@moz-extension://f3feffe7-1653-cd48-86d9-690b0072a16b/dist/js/polyfill.js:1:305
[1]</<@moz-extension://f3feffe7-1653-cd48-86d9-690b0072a16b/dist/js/polyfill.js:5:1
[1]<@moz-extension://f3feffe7-1653-cd48-86d9-690b0072a16b/dist/js/polyfill.js:2:2
s@moz-extension://f3feffe7-1653-cd48-86d9-690b0072a16b/dist/js/polyfill.js:1:246
e@moz-extension://f3feffe7-1653-cd48-86d9-690b0072a16b/dist/js/polyfill.js:1:425
@moz-extension://f3feffe7-1653-cd48-86d9-690b0072a16b/dist/js/polyfill.js:1:11
Line: 272, column: 3

**Comment by marcelklehr** [comment]
Ah, I'm sorry the wrong page is selected in the inspector by default: background.html. It should be options.html. You can change that in the top right corner, but before you open the inspector again, open the addon settings in the addon manager, so you can use the inspector without the options disappearing.

**Comment by bunducafe** [comment]
hm, either I am blind or stupid (maybe both). But I am not able to change something in the top right corner to options.html... how do I do that exactly?!

**Comment by marcelklehr** [comment]
Mh. looks like you ran the tests and they weren't cleaned up properly.

Open the addon debugger again, go to the scratchpad tab (if you can't see it, enable it in the options at the top right), paste the following and click 'Run':

```js
(async function () {
  await browser.storage.local.set({'accounts':{}})
})()
```

**Comment by bunducafe** [comment]
I am still kind of unsure, if what I am trying to do here is correct, but after putting this code into the scratch pad, that's what the console is telling me:

Unexpected invalid url: Scratchpad/1  webextension.js:312
	WebExtensionChildActor.prototype._allowSource resource://devtools/server/actors/webextension.js:312:5
	<anonym> self-hosted:991:17
	TabSources/this.allowSource resource://devtools/server/actors/utils/TabSources.js:32:39
	_addSource resource://devtools/server/actors/script.js:1952:10
	onNewScript resource://devtools/server/actors/script.js:1916:5
	onNewScript self-hosted:991:17
	evalWithDebugger resource://devtools/server/actors/webconsole.js:1423:16
	onEvaluateJS resource://devtools/server/actors/webconsole.js:931:20
	onEvaluateJSAsync resource://devtools/server/actors/webconsole.js:902:20
	onEvaluateJSAsync self-hosted:993:17
	onPacket resource://devtools/server/main.js:1788:15
	receiveMessage resource://devtools/shared/transport/transport.js:761:7

**Comment by marcelklehr** [comment]
@bunducafe That's a [bug in firefox](https://bugzilla.mozilla.org/show_bug.cgi?id=1412712). I think it should work in spite of the error. Could you try opening floccus options again?

**Comment by bunducafe** [comment]
@marcelklehr very strange... I have the exact same setup on my iMac and it seems that the Firefox instance here on the macbook is somehow broken. So, I will try to completely delete firefox from this machine and start from the scratch. Then I will drop a comment, whether the problem still persists or if it is magically solved.

**Comment by marcelklehr** [comment]
You might want to try creating a new profile first... :)

**Comment by bunducafe** [comment]
@marcelklehr I tried. New profile, unfortunately I cannot open the "Add Account" PopUp. When clicked on the Floccus icon the Add Acount Popup appears, but when trying to add an account nothing further happens.... It's a bit strange to me to be honest.

Another strange thing: I could run the test and it passed although I don't understand which credentials and which server the extension might be using because I could not fill mine within the new profile of FF.

**Comment by marcelklehr** [comment]
@bunducafe Someone else had the same issue recently, and apparently was able to resolve it by retrying many times: https://github.com/marcelklehr/floccus/issues/66

**Comment by marcelklehr** [comment]
If the Add account button is still an issue for you, could you provide logs as follows?

#### Firefox
* Go to `about:debugging`
* enable debugging
* click on `debug` button next to floccus entry
* go to floccus options and click `Add account`
* go back to the inspector window to the console tab and copy the log messages

**Comment by marcelklehr** [comment]
Nevermind. Fixed in v2.0.4 :) Thanks for your patience!

**Comment by bunducafe** [comment]
@marcelklehr Thank you... I will give it another try ;)
